devices/debian_wifi.py

```diff
@@ -1,88 +1,109 @@
-import re
+import re, time
 import debian
 import pexpect
+from countrycode import countrycode
+from lib.wifi import wifi_client_stub
 
-class DebianWifi(debian.DebianBox):
+class DebianWifi(debian.DebianBox, wifi_client_stub):
     '''Extension of Debian class with wifi functions'''
 
     model = ('debianwifi')
+    iface_wlan = "wlan0"
+    iface_wlan1 = "wlan1"
 
-    iface_wlan = "wlan1"
+    def disable_and_enable_wifi(self, iface):
+       self.set_link_state(iface,"down")
+       self.set_link_state(iface,"up")
 
-    def scan(self):
+    def disable_wifi(self, iface):
+       self.set_link_state(iface,"down")
+
+    def enable_wifi(self, iface):
+       self.set_link_state(iface,"up")
+
+    def wifi_scan(self, iface):
         from tests.lib.installers import install_iw
         install_iw(self)
 
-        self.sendline('iw %s scan | grep SSID:' % self.iface_wlan)
-        self.expect(self.prompt)
+        output = self.sudo_sendline('iw %s scan | grep SSID:' % iface)
+        return output
 
-        return self.before
-
-    def wpa_connect(self, ssid_name, password):
-         '''Generate WPA supplicant file and execute it'''
-         self.sendline("rm /etc/"+ssid_name+".conf")
-         self.expect(self.prompt)
-         self.sendline("wpa_passphrase "+ssid_name+ " >> /etc/"+ssid_name+".conf")
-         self.expect("")
-         self.sendline(password)
-         self.expect(self.prompt,timeout=20)
-         self.sendline("cat /etc/"+ssid_name+".conf")
-         self.expect(self.prompt)
-         self.sendline("wpa_supplicant  -B -Dnl80211 -i"+self.iface_wlan+ " -c/etc/"+ssid_name+".conf")
-         self.expect(self.prompt)
-         wpa_start = None
-         wpa_start = self.before
-         match = re.search('Successfully initialized wpa_supplicant',wpa_start)
-         if match:
-            return match.group(0)
-         else:
-            return None
+    def wifi_check_ssid(self, iface, ssid_name):
+        from tests.lib.installers import install_iw
+        install_iw(self)
 
-    def wlan_connectivity(self):
+        output = self.sudo_sendline('iw %s scan | grep "SSID: %s"' % (iface,ssid_name))
+        match = re.search("%s\"\s+.*(%s)"%(ssid_name,ssid_name),output)
+        if match:
+            return True
+        else:
+            return False
+
+    def wifi_connect(self, iface, ssid_name, password=None, security_mode=None):
```

Comment:
Why do you even need the password arg here? It should all be handled properly by sudo_sendline()

Reply:
This is the wifi password

Reply:
oh you are right, I was still stuck on sudo password... this is OK then.